java/vector/src/test/java/org/apache/arrow/vector/TestFixedSizeListVector.java

```diff
@@ -232,4 +235,51 @@ public void testConsistentChildName() throws Exception {
     }
   }
 
+  @Test
+  public void testUnionFixedSizeListWriter() throws Exception {
+    try (final FixedSizeListVector vector1 = FixedSizeListVector.empty("vector", 3, allocator)) {
+
+      UnionFixedSizeListWriter writer1 = vector1.getWriter();
+      writer1.allocate();
+
+      //set some values
+      writeListVector(writer1, new int[] {1, 2, 3});
+      writeListVector(writer1, new int[] {4, 5, 6});
+      writeListVector(writer1, new int[] {7, 8, 9});
+      writer1.setValueCount(3);
+
+      assertEquals(3, vector1.getValueCount());
```

Comment:
is there a way to avoid string comparison?

Reply:
Yes, fixed now.